runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -2088,8 +2128,10 @@ TR_VectorAPIExpansion::transformIL(bool checkBoxing)
 
                   if (!vectorizedOrScalarized)
                      {
-                     TR_ASSERT_FATAL(operand->getOpCodeValue() == TR::aload || operand->getOpCodeValue() == TR::acall,
-                                     "Operand can only be aload or acall");
+                     TR_ASSERT_FATAL(operand->getOpCodeValue() == TR::aload ||
+                                     operand->getOpCodeValue() == TR::acall ||
+                                     operand->getOpCodeValue() == TR::New,
+                                     "Operand can only be aload or acall (child %d)", i);
```

Comment:
The message still claims that `New` isn't allowed

At a glance it doesn't seem like unboxing requires the `operand` to have such a specific opcode. Is this assertion perhaps too strict? (even now that it accepts `New`)

Reply:
Yeah, I think the idea was to make sure we are looking at the right operands of the intrinsic. We can just check for the TR::Address type but I think it's better to do it in another PR.